kyuubi-server/src/main/scala/org/apache/kyuubi/server/statestore/api/Metadata.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.statestore.api
+
+/**
+ * The metadata to store. It including three parts:
+ * 1. session related metadata.
+ * 2. request related metadata.
+ * 3. engine related metadata.
+ *
+ * @param identifier the identifier id.
+ * @param username the user name.
+ * @param ipAddress the ip address.
+ * @param kyuubiInstance the kyuubi instance.
+ * @param state the state.
+ * @param resource the main resource.
+ * @param className the main class name.
+ * @param requestName the request name.
+ * @param requestConf the request config map.
+ * @param requestArgs the request arguments.
+ * @param createTime the create time.
+ * @param engineType the engine type.
+ * @param engineId the engine id.
+ * @param engineName the engine name.
+ * @param engineUrl the engine tracking url.
+ * @param engineState the engine state.
+ * @param engineError the engine error diagnose.
+ * @param endTime the end time.
+ */
+case class Metadata(
+    identifier: String,
+    realUser: String = null,
+    username: String = null,
+    ipAddress: String = null,
+    kyuubiInstance: String = null,
+    state: String = null,
+    resource: String = null,
+    className: String = null,
+    requestName: String = null,
+    requestConf: Map[String, String] = Map.empty,
+    requestArgs: Seq[String] = Seq.empty,
+    createTime: Long = 0L,
+    engineType: String = null,
```

Comment:
maybe we need identify the session type according to engineType.

Reply:
For `SPARK_SQL, FLINK_SQL, TRINO, HIVE_SQL, JDBC`, it is SQL session.

And for `SPARK, FLINK`, it is batch session.

Reply:
For `SPARK_SQL, FLINK_SQL, TRINO, HIVE_SQL, JDBC`, it is SQL session.

And for `SPARK, FLINK`, it is batch session.